python/cuml/common/base.py

```diff
@@ -482,6 +449,6 @@ def _determine_stateless_output_type(output_type, input_obj):
 
     # If we are using 'input', determine the the type from the input object
     if temp_output == 'input':
-        temp_output = _input_to_type(input_obj)
+        temp_output = cuml.common.input_utils.determine_array_type(input_obj)
```

Comment:
I mentioned this above, but this is to avoid circular import errors. There are a lot of inter dependencies between the modules in `cuml.common` that make it difficult to do `from ... import ...`.

We might be able to resolve this issue by moving some of the functions into different modules (for example, separating the array type utility functions from `input_to_cuml_array`). But that would add more diffs to this PR which I was avoiding.